internal/servers/worker/tcp_proxy.go

```diff
@@ -65,13 +65,16 @@ func (w *Worker) handleTcpProxyV1(connCtx context.Context, clientAddr *net.TCPAd
 	go func() {
 		defer connWg.Done()
 		_, err := io.Copy(netConn, tcpRemoteConn)
+		netConn.Close()
```

Comment:
I am not sure but seems like a potential race between the `Close` calls and the `io.Copy` calls in the two `go func()`.  Would we not want to rather do the two closes after the `congWg.Wait()`, therefore giving both anon funcs time to perform the `io.Copy()` as needed?

Reply:
Copy blocks until EOF. So what this is doing is saying, if either end EOFs, ensure that the other end also exits. This is how things work on the client side. I added it here too because I have a sneaking suspicion that this might be the cause of a couple instances we've seen where a client disconnects but the session doesn't end -- the EOF on one side doesn't end up resulting in the other side being closed properly. I have to test to see if it resolves that, but in the meantime it seemed like a good thing to add anyways.

Reply:
Aha I see what you are doing here now, thanks!